src/main/java/htsjdk/utils/ValidationUtils.java

```diff
@@ -0,0 +1,150 @@
+ * THE SOFTWARE.
+ */
+
+package htsjdk.utils;
+
+import java.util.Collection;
+import java.util.function.Supplier;
+
+/**
+ * Simple functions that streamline the checking of values.
+ */
+public class ValidationUtils {
+    /**
+     * Checks that an Object {@code object} is not null and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object) {
+        return ValidationUtils.nonNull(object, "Null object is not allowed here.");
+    }
+
+    /**
+     * Checks that an {@link Object} is not {@code null} and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @param message the text message that would be passed to the exception thrown when {@code o == null}.
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object, final String message) {
+        if (object == null) {
+            throw new IllegalArgumentException(message);
+        }
+        return object;
+    }
+
+    /**
+     * Checks that an {@link Object} is not {@code null} and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @param message the text message that would be passed to the exception thrown when {@code o == null}.
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object, final Supplier<String> message) {
+        if (object == null) {
+            throw new IllegalArgumentException(message.get());
+        }
+        return object;
+    }
+
+
+    /**
+     * Checks that a {@link Collection} is not {@code null} and that it is not empty.
+     * If it's non-null and non-empty it returns the input, otherwise it throws an {@link IllegalArgumentException}
+     * @param collection any Collection
+     * @param message a message to include in the output
+     * @return the original collection
+     * @throws IllegalArgumentException if collection is null or empty
+     */
+    public static <I, T extends Collection<I>> T nonEmpty(T collection, String message){
+        nonNull(collection, "The collection is null: " + message);
+        if(collection.isEmpty()){
+            throw new IllegalArgumentException("The collection is empty: " + message);
+        } else {
+            return collection;
+        }
+    }
+
+    /**
+     * Checks that a {@link Collection} is not {@code null} and that it is not empty.
+     * If it's non-null and non-empty it returns the true
+     * @param collection any Collection
+     * @return true if the collection exists and has elements
+     */
+    public static boolean isNonEmpty(Collection<?> collection){
```

Comment:
Seems confusing to me to have two different styles of validation here that do pretty much the same thing.